File: 0x17-doubly_linked_lists/palin.c
#include <stdio.h>

int is_palindrome(int n) {
    int original = n;
    int reversed = 0;

    while (n > 0) {
        reversed = reversed * 10 + n % 10;
        n /= 10;
    }

    return original == reversed;
}

void find_largest_palindrome() {
    int largest_palindrome = 0;

    for (int i = 100; i < 1000; ++i) {
        for (int j = 100; j < 1000; ++j) {
            int product = i * j;
            if (is_palindrome(product) && product > largest_palindrome) {
                largest_palindrome = product;
            }
        }
    }

    FILE *file = fopen("102-result", "w");
    if (file != NULL) {
        fprintf(file, "%d\n", largest_palindrome);
        fclose(file);
    } else {
        printf("Error opening file.\n");
    }
}

int main(void) {
    find_largest_palindrome();
    return 0;
}

File: 0x17-doubly_linked_lists/palin.py
def is_palindrome(n):
    return str(n) == str(n)[::-1]

def find_largest_palindrome():
    largest_palindrome = 0
    factors = (0, 0)

    for i in range(100, 1000):
        for j in range(100, 1000):
            product = i * j
            if is_palindrome(product) and product > largest_palindrome:
                largest_palindrome = product
                factors = (i, j)

    return largest_palindrome, factors

result, factors = find_largest_palindrome()

with open("102-result", "w") as file:
    file.write(f"{result} = {factors[0]} x {factors[1]}\n")
